dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/strategy/FileAssetViewStrategy.java

```diff
@@ -0,0 +1,117 @@
+import com.dotmarketing.portlets.contentlet.business.ContentletCache;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.fileassets.business.FileAsset;
+import com.dotmarketing.util.Logger;
+import com.liferay.portal.model.User;
+import java.util.Map;
+import java.util.Set;
+
+public class FileAssetViewStrategy extends WebAssetStrategy<FileAsset> {
+
+    private final ContentletCache contentletCache;
+
+    /**
+     * Main Constructor
+     * @param toolBox
+     */
+    public FileAssetViewStrategy(final TransformToolbox toolBox) {
+        super(toolBox);
+        contentletCache = CacheLocator.getContentletCache();
+    }
+
+    /**
+     * fromContentlet retrieves the concrete specific type of contentlet
+     * @param contentlet
+     * @return
+     */
+    @Override
+    public FileAsset fromContentlet(final Contentlet contentlet) {
+        final Contentlet cachedContent = contentletCache.get(contentlet.getInode());
+        if(cachedContent instanceof FileAsset){
+           return (FileAsset)cachedContent;
+        }
+        return toolBox.fileAssetAPI.fromContentlet(contentlet);
+    }
+
+    /**
+     * Transform entry point
+     * @param fileAsset
+     * @param map
+     * @param options
+     * @param user
+     * @return
+     * @throws DotDataException
+     */
+    @Override
+    protected Map<String, Object> transform(final FileAsset fileAsset,
+            final Map<String, Object> map,
+            final Set<TransformOptions> options, final User user) throws DotDataException {
+
+        if(isNotSet(fileAsset.getTitle())) {
+            final Identifier identifier = toolBox.identifierAPI.find(fileAsset.getIdentifier());
+            map.put(TITLE_FIELD, identifier.getAssetName());
+        }
+
+        final String fileName = fileAsset.getFileName();
+        final String underlyingFileName = fileAsset.getUnderlyingFileName();
+        final long fileSize = fileAsset.getFileSize();
+        map.put(MIMETYPE_FIELD, fileAsset.getMimeType());
+        map.put("name", fileName);
+        map.put("size", fileSize);
+        final String description = fileAsset.getStringProperty(DESCRIPTION);
+        map.put(DESCRIPTION, isSet(description) ? description : BLANK );
+        map.put(UNDERLYING_FILENAME, underlyingFileName);
+        if(options.contains(LOAD_META)) {
+            map.put(META_DATA_FIELD, fileAsset.getMetaData());
+        }
+        map.put("path", fileAsset.getPath());
+        final String parent = fileAsset.getParent();
+        map.put("parent",  isSet(parent) ? parent : BLANK );
+
+        if(isImage(underlyingFileName)) {
+            map.put("width", fileAsset.getWidth());
+            map.put("height", fileAsset.getHeight());
+        }
+
+        map.put("extension", getFileExtension(underlyingFileName));
+        try {
+            map.put("__icon__", getIconClass(fileAsset));
+        }catch (Exception e){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)